analyzers/tests/SonarAnalyzer.TestFramework/Common/SourceGeneratorProvider.cs

```diff
@@ -26,13 +26,34 @@ namespace SonarAnalyzer.TestFramework.Common;
 public static class SourceGeneratorProvider
 {
     private static readonly string RazorSourceGeneratorPath =
-        Path.Combine(Path.GetDirectoryName(typeof(SourceGeneratorProvider).Assembly.Location), "Dependencies", "Microsoft.CodeAnalysis.Razor.Compiler.SourceGenerators.dll");
+        Path.Combine(LatestSdkFolder(), "Sdks", "Microsoft.NET.Sdk.Razor", "source-generators", "Microsoft.CodeAnalysis.Razor.Compiler.SourceGenerators.dll");
 
     public static AnalyzerFileReference[] SourceGenerators { get; } =
     [
-        new(RazorSourceGeneratorPath, new AssemblyLoader())
+        new(CheckRazorSourceGeneratorPath(), new AssemblyLoader())
     ];
 
+    public static string CheckRazorSourceGeneratorPath() =>
+        File.Exists(RazorSourceGeneratorPath) ? RazorSourceGeneratorPath : throw new FileNotFoundException($"Razor sourcegenerator not found: {RazorSourceGeneratorPath}");
+
+    public static string LatestSdkFolder()
+    {
+        var objectAssembly = typeof(object).Assembly;
+        var objectAssemblyDirectory = new FileInfo(objectAssembly.Location).Directory; // C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.4
+        var dotnetDirectory = objectAssemblyDirectory.Parent.Parent.Parent;            // C:\Program Files\dotnet
+        var sdkDirectory = Path.Combine(dotnetDirectory.FullName, "sdk");              // C:\Program Files\dotnet\sdk
+        if (!Directory.Exists(sdkDirectory))
+        {
+            throw new NotSupportedException($"Razor analysis is only supported for .Net Core.");
```

Comment:
```suggestion
            throw new NotSupportedException($"'{sdkDirectory}' does not exist. Ensure you are using .NET (Core).");
```
or
```suggestion
            throw new NotSupportedException($"Razor analysis is only supported for .Net Core. (Could not find SDK at '{sdkDirectory}'.)");
```

Reply:
I improved/extended a bit the exception message 👍